OpenRA.Game/GameRules/Ruleset.cs

```diff
@@ -81,6 +81,19 @@ public class Ruleset
 							throw new YamlException("Weapon type {0}: {1}".F(weapon.Key, e.Message));
 						}
 					}
+
+					var projectileLoaded = weapon.Value.Projectile as IRulesetLoaded<WeaponInfo>;
```

Comment:
I would consider this PRed to upstream at a point. This doesn't seem to be a hack so it should be able to be accepted.